### PR TITLE
DM-22364: Now must explicitly specify where a fake posix datastore exists

### PR DESCRIPTION
**Pull request by timj**

### CODE REVIEW

File: tests/test_cmdLineFwk.py [hunk 1]
```diff
@@ -126,10 +126,11 @@ def _makeArgs(pipeline=None, qgraph=None, pipeline_actions=(), order_pipeline=Fa
     pipeline_dot : `str`
         Name of the DOT file to write pipeline graph.
     qgraph_dot : `str`
-        Name of the DOT file to write QGrpah representation.
+        Name of the DOT file to write QGraph representation.
     """
     args = argparse.Namespace()
-    args.butler_config = None
+    args.butler_config = Config()
+    args.butler_config.configFile = "."  # Lie about where the datastore is located
```
Comment (andy-slac) [inline tests/test_cmdLineFwk.py]: I'm against lying in general. Could you rephrase it to something like "configFile must be specified but it does not matter in this case"? Could you specify `configFile` in a constructor? If empty `Config` constructor is not useful for cases like this maybe it should be disabled entirely?

Comment (timj) [reply tests/test_cmdLineFwk.py]: In this particular case you only want access to registry and don't care where datastore is located. An empty Config to construct a Butler in theory could be fine since defaults are picked up, but in the specific case the default is to use a relocatable datastore configuration so you then have a bit of a chicken and egg problem. I don't think it's the fault of Config that it's not usable to construct a Butler if it's empty.

Comment (andy-slac) [reply tests/test_cmdLineFwk.py]: OK to merge.
